cmd/celestia-appd/cmd/modify_root_command_multiplexer.go

```diff
@@ -33,22 +32,41 @@ func modifyRootCommand(rootCommand *cobra.Command) {
 		panic(err)
 	}
 
-	var extraArgs []string
-	if v2UpgradeHeight != "" {
-		extraArgs = append(extraArgs, "--v2-upgrade-height="+v2UpgradeHeight)
+	var (
+		embeddedArgsFile string
+		embeddedArgs     EmbeddedArgs
+	)
+	// get the home directory from the flag
+	args := os.Args
+	for i := 0; i < len(args); i++ {
+		if args[i] == fmt.Sprintf("--%s", FlagEmbeddedArgsFile) && i+1 < len(args) {
+			embeddedArgsFile = filepath.Clean(args[i+1])
+		} else if strings.HasPrefix(args[i], fmt.Sprintf("--%s=", FlagEmbeddedArgsFile)) {
+			embeddedArgsFile = filepath.Clean(args[i][7:])
```

Comment:
This code makes the assumption that we'll never change the string inside FlagEmbeddedArgsFile. Can this please be made more robust? Ideally it should not use a hard-coded index like `7`.

```go
		} else if strings.HasPrefix(args[i], fmt.Sprintf("--%s=", FlagEmbeddedArgsFile)) {
			trimmed := strings.TrimPrefix(args[i], fmt.Sprintf("--%s=", FlagEmbeddedArgsFile))
			return filepath.Clean(trimmed)
		}
```


[blocking] needs unit tests